### PR TITLE
added a note about experimental features

### PR DESCRIPTION
**Pull request by fabpot**
I think `master` makes sense here as no experimental features can be added anymore in earlier versions.

### CODE REVIEW

**Comment by fabpot** [comment]
http://symfony.com/blog/experimental-features for more information about this change.

File: contributing/community/releases.rst [hunk 1]
```diff
@@ -150,6 +150,10 @@ of Symfony to the next one.
 Whenever keeping backward compatibility is not possible, the feature, the
 enhancement or the bug fix will be scheduled for the next major version.
 
+However, features marked with the ``@experimental`` tag are **not** part of the
+backward compatibility promise and their APIs can change from one minor version
```
Comment (greg0ire) [inline contributing/community/releases.rst]: "backwards" misses its 's'

File: contributing/code/bc.rst [hunk 2]
```diff
@@ -32,6 +32,14 @@ The second section, "Working on Symfony Code", is targeted at Symfony
 contributors. This section lists detailed rules that every contributor needs to
 follow to ensure smooth upgrades for our users.
 
+.. warning::
+
+    Code marked with the ``@experimental`` or the ``@internal`` tags are
+    excluded from our Backwards Compatibility promise.
+
+    Also note that backward compatibility breaks are tolerated if they help fix
```
Comment (greg0ire) [inline contributing/code/bc.rst]: The "s" misses here too (I think it is actually correct, but it seems to be written with an "s" [in the bc promise doc](https://symfony.com/doc/current/contributing/code/bc.html))

Comment (ogizanagi) [inline contributing/code/bc.rst]: Your reference seems to state `backward` without the trailing `s` is the most appropriate as an adjective:

> If you’re using British English, it’s typically backwards as an adverb and backward as an adjective, so you have two things to remember.

So maybe the BC promise doc should be updated instead 😄

Comment (greg0ire) [inline contributing/code/bc.rst]: > So maybe the BC promise doc should be updated instead 😄

So should the first sentence of this PR then :P

File: contributing/code/bc.rst [hunk 3]
```diff
@@ -32,6 +32,14 @@ The second section, "Working on Symfony Code", is targeted at Symfony
 contributors. This section lists detailed rules that every contributor needs to
 follow to ensure smooth upgrades for our users.
 
+.. warning::
+
+    Code marked with the ``@experimental`` or the ``@internal`` tags are
```
Comment (ogizanagi) [inline contributing/code/bc.rst]:
```diff
- are
+ is
```

?

**Comment by kherge** [comment]
Why not place experimental features in a separate repository?

Once they're finalized, they can then be merged into the main repository with commit history intact.

**Comment by fabpot** [comment]
@kherge Because most people won't use them. Experimental features need feedback, and having them in core helps a lot.

**Review by javiereguiluz — APPROVED** [review]
:+1:

There were somme comments about "backward compat..." vs "backwardS compat..."  Let's ask our native English friend @weaverryan about this. Thanks!

File: contributing/code/bc.rst [hunk 4]
```diff
@@ -32,6 +32,14 @@ The second section, "Working on Symfony Code", is targeted at Symfony
...
+    excluded from our Backwards Compatibility promise.
+
+    Also note that backward compatibility breaks are tolerated if they help fix
+    a security issue.
```
Comment (nicolas-grekas) [inline contributing/code/bc.rst]: if they help fix a security issue => if they are required to fix a security issue?

File: contributing/community/releases.rst [hunk 5]
```diff
@@ -150,6 +150,10 @@ of Symfony to the next one.
...
+to the next. The changelog must explain the changes and how to upgrade.
```
Comment (nicolas-grekas) [inline contributing/community/releases.rst]: s/must/should/? It's not always worth the burden (eg we didn't follow that "must" for LDAP, and it was fine?)

**Review by nicolas-grekas — COMMENTED** [review]
- Should we add a note about `@final` also?
- What about experimental feats + LTS? They look contradictory to me. Should we have a written policy saying we don't add experimental feats to LTS?
- How long can a feat be marked as experimental? Only for one release to me - the next one should either drop, release as stable, or at least update the `@experimental` tag to explicitly mention the extended laboratory period?
In this spirit, I used `@experimental in version 3.3` annotations in https://github.com/symfony/symfony/pull/20973.

File: contributing/community/releases.rst (hunk 1 above)
Comment (Wirone) [inline contributing/community/releases.rst]: Misses on purpose - there's nothing like "backwards compatibility".

Comment (greg0ire) [inline contributing/community/releases.rst]: @Wirone : if that's true the occurences of "backwards compatibility" in this PR should be removed.

Comment (fabpot) [inline contributing/community/releases.rst]: see #7462 to fix existing occurrences.

File: contributing/code/bc.rst (hunk 3 above)
Comment (fabpot) [inline contributing/code/bc.rst]: fixed

File: contributing/code/bc.rst (hunk 4 above)
Comment (fabpot) [inline contributing/code/bc.rst]: fixed

File: contributing/community/releases.rst (hunk 5 above)
Comment (fabpot) [inline contributing/community/releases.rst]: That's a must. We didn't do that for LDAP and people now are confused.

**Comment by fabpot** [comment]
I've created a new document on experimental features to add some more details.

File: contributing/code/experimental.rst [hunk 6]
```diff
@@ -0,0 +1,23 @@
+Experimental Features
+=====================
+
+All Symfony features benefits from our :doc:`Backward Compatibility Promise
```
Comment (javiereguiluz) [inline contributing/code/experimental.rst]: `benefits` -> `benefit`

File: contributing/code/experimental.rst [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+</contributing/code/bc>` to give developers the confidence to upgrade to new
+versions safely and more often.
+
+But sometimes, a new feature is controversial. Or finding a good API is not
+easy. In such cases, we want to be able to gather feedback from real-world
```
Comment (javiereguiluz) [inline contributing/code/experimental.rst]: `we want to be able to gather feedback` -> `we prefer to gather feedback` ?

File: contributing/code/experimental.rst [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+usage, adapt the API, or remove it altogether. Doing so is not possible with a
+no BC-break approach.
+
+To avoid being bound to our backward compatibility promise, such features can
+be marked as **experimental**; classes and methods parts of an experimental
+feature must be marked with the ``@experimental`` tag.
```
Comment (javiereguiluz) [inline contributing/code/experimental.rst]: `...; classes and methods parts of an experimental feature must be marked ...` -> `... and their classes and methods must be marked ...` ?

File: contributing/code/experimental.rst [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+
+A feature can be marked as being experimental for only one minor version, and
+can never be introduced in an :ref:`LTS version <releases-lts>`. The core team
```
Comment (javiereguiluz) [inline contributing/code/experimental.rst]: `in an` -> `in a`

**Review by javiereguiluz — APPROVED** [review]
👍  nice!

File: contributing/code/experimental.rst (hunk 6 above)
Comment (fabpot) [inline contributing/code/experimental.rst]: fixed

File: contributing/code/experimental.rst (hunk 7 above)
Comment (fabpot) [inline contributing/code/experimental.rst]: sure

File: contributing/code/experimental.rst (hunk 8 above)
Comment (fabpot) [inline contributing/code/experimental.rst]: much better indeed

File: contributing/code/experimental.rst (hunk 9 above)
Comment (fabpot) [inline contributing/code/experimental.rst]: fixed

**Pull request merged**